assets/js/blocks/reviews/frontend-container-block.tsx

```diff
@@ -4,51 +4,71 @@
 import { __, _n, sprintf } from '@wordpress/i18n';
 import { speak } from '@wordpress/a11y';
 import { Component } from '@wordpress/element';
-import PropTypes from 'prop-types';
 
 /**
  * Internal dependencies
  */
 import { getSortArgs } from './utils';
 import FrontendBlock from './frontend-block';
+import { ReviewBlockAttributes } from './attributes';
+import { Review } from '../../base/components/reviews/types';
+
+type FrontendContainerBlockProps = {
+	attributes: ReviewBlockAttributes;
+};
 
 /**
  * Container of the block rendered in the frontend.
  */
-class FrontendContainerBlock extends Component {
-	constructor() {
-		super( ...arguments );
+class FrontendContainerBlock extends Component<
+	FrontendContainerBlockProps,
+	{ orderby: string; reviewsToDisplay: number }
+> {
+	constructor( props: FrontendContainerBlockProps ) {
+		super( props );
 		const { attributes } = this.props;
 
 		this.state = {
-			orderby: attributes.orderby,
-			reviewsToDisplay: parseInt( attributes.reviewsOnPageLoad, 10 ),
+			orderby: attributes?.orderby,
+			reviewsToDisplay: this.getReviewsOnPageLoad(),
 		};
 
 		this.onAppendReviews = this.onAppendReviews.bind( this );
 		this.onChangeOrderby = this.onChangeOrderby.bind( this );
 	}
 
-	onAppendReviews() {
+	getReviewsOnPageLoad() {
+		const { attributes } = this.props;
+
+		return typeof attributes.reviewsOnPageLoad === 'number'
```

Comment:
The original implementation is calling `parseInt` which expects a number. I'm not really worried about that since parseInt when passing a number returns the number itself, but TS says it has to be a string, so for both these attributes I'm checking their type first instead so we always return a number.